File: Code/Lib/Xib Helper/UIView+XIBEdit.m
//
//  UIView+XIBEdit.m
//  client
//
//  Created by ZHAOCHENJUN on 15/5/31.
//  Copyright (c) 2015年 zhiqin. All rights reserved.
//

#import "UIView+XIBEdit.h"

@implementation UIView (UIViewExtend)

@dynamic borderColor,borderWidth,cornerRadius,shadowColor,shadowOffset,shadowRadius;

/*
// Only override drawRect: if you perform custom drawing.
// An empty implementation adversely affects performance during animation.
- (void)drawRect:(CGRect)rect {
    // Drawing code
}
*/

-(void)setBorderColor:(UIColor *)borderColor{
	[self.layer setBorderColor:borderColor.CGColor];
}

-(void)setBorderWidth:(NSInteger)borderWidth{
	[self.layer setBorderWidth:borderWidth];
}

-(void)setCornerRadius:(NSInteger)cornerRadius{
	[self.layer setCornerRadius:cornerRadius];
}

-(void)setShadowColor:(UIColor *)shadowColor{
	[self.layer setShadowColor:shadowColor.CGColor];
}

-(void)setShadowOffset:(CGSize)shadowOffset{
	[self.layer setShadowOffset:shadowOffset];
}

-(void)setShadowRadius:(NSInteger)shadowRadius{
	[self.layer setShadowRadius:shadowRadius];
}


@end

File: Code/Lib/Xib Helper/UIView+XIBEdit.swift
//
//  UIView+XIBEdit.swift
//  NewbieZone
//
//  Created by zhaochenjun on 15/12/29.
//  Copyright © 2015年 kazmastudio. All rights reserved.
//

import UIKit

extension UIView {
    
    @IBInspectable var cornerRadius: CGFloat {
        
        get {
            
            return layer.cornerRadius
        
        }
        
        set {
        
            layer.cornerRadius = newValue
            layer.masksToBounds = newValue > 0
        
        }
    }
    
    @IBInspectable var shadowOffset: CGSize {
        
        get {
            
            return layer.shadowOffset
            
        }
        
        set {
            layer.shadowRadius = 2
            layer.shadowOpacity = 0.3
            layer.shadowOffset = newValue
            layer.masksToBounds = false
            
        }
    }
    
    @IBInspectable var borderWidth: CGFloat {
        
        get {
        
            return layer.borderWidth
        
        }
        
        set {
        
            layer.borderWidth = newValue
            layer.masksToBounds = newValue > 0
        }
    }
    
    
}